src/analysis/special_functions/trigonometric.lean

```diff
@@ -1748,15 +1750,105 @@ by simp only [tan_eq_sin_div_cos]; exact
+  ext x,
+  rw tan_eq_sin_div_cos x,
+  ring,
+end
+
+lemma tendsto_sin_neg_pi_div_two : tendsto sin (𝓝[Ioi (-(π/2))] (-(π/2))) (𝓝 (-1)) :=
+by { convert continuous_sin.continuous_within_at, simp }
+
+lemma tendsto_cos_neg_pi_div_two : tendsto cos (𝓝[Ioi (-(π/2))] (-(π/2))) (𝓝[Ioi 0] 0) :=
+begin
+  apply tendsto_nhds_within_of_tendsto_nhds_of_eventually_within,
+  { convert continuous_cos.continuous_within_at, simp },
+  { filter_upwards [Ioo_mem_nhds_within_Ioi (set.left_mem_Ico.mpr (norm_num.lt_neg_pos
+      _ _ pi_div_two_pos pi_div_two_pos))] λ x hx, cos_pos_of_mem_Ioo hx.1 hx.2 },
+end
+
+lemma tendsto_tan_neg_pi_div_two : tendsto tan (𝓝[Ioi (-(π/2))] (-(π/2))) at_bot :=
+begin
+  convert tendsto_mul_at_bot (by norm_num) (tendsto.inv_tendsto_zero tendsto_cos_neg_pi_div_two)
+            tendsto_sin_neg_pi_div_two,
+  ext x,
+  rw tan_eq_sin_div_cos x,
+  ring,
+end
+
+/-!
+### Continuity and differentiability of arctan
+
+The continuity of `arctan` is difficult to prove due to `arctan` being (indirectly) defined naively
+via `classical.some`. The proof therefore uses the general theorem that monotone functions are
+homeomorphisms: `homeomorph_of_strict_mono_continuous_Ioo`. We first prove that `tan` (restricted)
+is a homeomorphism whose inverse is definitionally equal to `arctan`. The fact that `arctan` is
+continuous is then derived from the fact that it is equal to a homeomorphism, and its
+differentiability is in turn derived from its continuity using `has_deriv_at.of_local_left_inverse`.
+-/
+
+/-- The function `tan`, restricted to the open interval (-π/2, π/2), is a homeomorphism. The
+inverse function of that homeomrphism is definitionally equal to `arctan` via `homeomorph.change_inv`.
+-/
+def tan_homeomorph : (Ioo (-(π/2)) (π/2)) ≃ₜ ℝ :=
+(homeomorph_of_strict_mono_continuous_Ioo tan (by linarith [pi_div_two_pos])
+  (λ x y, tan_lt_tan_of_lt_of_lt_pi_div_two) continuous_on_tan_Ioo tendsto_tan_pi_div_two
+    tendsto_tan_neg_pi_div_two).change_inv (λ x, ⟨arctan x, arctan_mem_Ioo x⟩) tan_arctan
+
+lemma tan_homeomorph_inv_fun_eq_arctan : coe ∘ tan_homeomorph.inv_fun = arctan := rfl
+
+lemma continuous_arctan : continuous arctan :=
+continuous_subtype_coe.comp tan_homeomorph.continuous_inv_fun
+
+lemma has_deriv_at_arctan (x:ℝ) : has_deriv_at arctan (1 / (1 + x^2)) x :=
+begin
+  have h1 : 0 < 1 + x^2 := by nlinarith,
+  have h2 : cos (arctan x) ≠ 0 := by { rw cos_arctan, exact ne_of_gt (one_div_pos.mpr (sqrt_pos.mpr h1)) },
+  simpa [(cos_arctan x), sqr_sqrt (le_of_lt h1)] using has_deriv_at.of_local_left_inverse
+    continuous_arctan.continuous_at (has_deriv_at_tan (cos_ne_zero_iff.mp h2))
+      (one_div_ne_zero (pow_ne_zero 2 h2)) (by {apply eventually_of_forall, exact tan_arctan} ),
+end
+
+lemma differentiable_at_arctan (x:ℝ) : differentiable_at ℝ arctan x :=
+(has_deriv_at_arctan x).differentiable_at
+
+@[simp] lemma deriv_arctan (x:ℝ) : deriv arctan x = 1 / (1 + x^2) :=
+(has_deriv_at_arctan x).deriv
+
```

Comment:
You are missing a few lemmas, to make sure that `simp` is able to compute the derivative of a function involving `arctan`. (Have a look in `test/differentiable.lean`). You can copy how this is done for `real.cos` between lines 355 and 388 in this file, replacing `cos` with `arctan` and the obvious adjustements.